networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/topics/validation/SignedAggregateAndProofValidator.java

```diff
@@ -71,8 +79,13 @@ public InternalValidationResult validate(final ValidateableAttestation attestati
     final AggregatorIndexAndEpoch aggregatorIndexAndEpoch =
         new AggregatorIndexAndEpoch(
             aggregateAndProof.getIndex(), compute_epoch_at_slot(aggregateSlot));
-    if (receivedValidAggregations.contains(aggregatorIndexAndEpoch)) {
-      LOG.trace("Rejecting duplicate aggregate");
+    if (receivedAggregatorIndexAndEpochs.contains(aggregatorIndexAndEpoch)) {
+      LOG.trace("Ignoring duplicate aggregate");
+      return IGNORE;
+    }
+
+    if (receivedValidAggregations.contains(attestation.hash_tree_root())) {
+      LOG.trace("Ignoring duplicate aggregate");
```

Comment:
nit: we should differentiate this trace message from the duplicate based on index/epoch check. If we're trying to track down why an attestation wasn't gossiped, knowing if it was rejected because it was identical to another attestation or just from the same index/epoch is very useful info.
```suggestion
      LOG.trace("Ignoring duplicate aggregate based on hash tree root");
```

Reply:
That's true. Made the change.